src/universal-dapp.js

```diff
@@ -700,7 +718,12 @@ UniversalDApp.prototype.runTx = function (data, args, cb) {
         transactions: [],
         uncleHeaders: []
       });
-      self.vm.runTx({block: block, tx: tx, skipBalance: true, skipNonce: true}, cb);
+      self.vm.runTx({block: block, tx: tx, skipBalance: true, skipNonce: true}, function (err, result) {
+        if (self.vm) {
+          result.transactionHash = self.transactionDebugger.web3().releaseCurrentHash(); // used to keep track of the transaction
```

Comment:
web3() is returning a web3 provider (the one loaded here is a custom provider that rely on the VM).
this hash is used to keep track of the transaction so that it is possible to retrieve it later (for now the instance of the  VM does not include any memory support). but it will certainly come later.